src/fondant/pipeline/pipeline.py

```diff
@@ -319,11 +333,193 @@ def __init__(
         self._graph: t.OrderedDict[str, t.Any] = OrderedDict()
         self.task_without_dependencies_added = False
 
+    def _build_component_op(
+        self,
+        name,
+        *,
+        arguments: t.Optional[t.Dict[str, t.Any]] = None,
+        input_partition_rows: t.Optional[t.Union[str, int]] = None,
+        cache: t.Optional[bool] = True,
+        cluster_type: t.Optional[str] = "default",
+        client_kwargs: t.Optional[dict] = None,
+        resources: t.Optional[Resources] = None,
+        schema: t.Optional[t.Dict[str, t.Any]] = None,
+        consumes: t.Optional[t.Dict[str, t.Any]] = None,
+        produces: t.Optional[t.Dict[str, t.Any]] = None,
+    ) -> ComponentOp:
+        """Building ComponentOp."""
+        if not self._is_custom_component(path_or_name=name):
+            name = self._get_registry_path(name)
+        return ComponentOp(
+            name,
+            arguments=arguments,
+            input_partition_rows=input_partition_rows,
+            cache=cache,
+            cluster_type=cluster_type,
+            client_kwargs=client_kwargs,
+            resources=resources,
+            schema=schema,
+            consumes=consumes,
+            produces=produces,
+        )
+
+    def read(
+        self,
+        name,
+        *,
+        arguments: t.Optional[t.Dict[str, t.Any]] = None,
+        input_partition_rows: t.Optional[t.Union[str, int]] = None,
+        cache: t.Optional[bool] = True,
+        cluster_type: t.Optional[str] = "default",
+        client_kwargs: t.Optional[dict] = None,
+        resources: t.Optional[Resources] = None,
+        schema: t.Dict[str, str],
```

Comment:
Shouldn't this schema be `Dict[str, Type]`? I would add some example on how this would look like in the docstring so the users know what to input and mainly because it's different from the `write` schema (this one would also benefit from having examples in the description)